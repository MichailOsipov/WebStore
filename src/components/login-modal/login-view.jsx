import * as React from 'react';
import {Button, Typography} from '@material-ui/core';
import {LoginModalForm} from './login-modal-form';

export const LoginView = ({onLogin, onGoMainPage}) => (
    <div>
        <Typography variant="h6">Авторизация</Typography>
        <LoginModalForm />
        <Button variant="contained" onClick={onLogin}>Логин</Button>
        <Button onClick={onGoMainPage}>Назад</Button>
    </div>
);
