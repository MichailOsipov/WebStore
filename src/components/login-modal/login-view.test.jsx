import * as React from 'react';
import {shallow} from 'enzyme';
import {LoginView} from './login-view';

describe('LoginView', () => {
    test('rendering', () => {
        const wrapper = shallow(
            <LoginView
                onLogin={jest.fn()}
                onGoMainPage={jest.fn()}
            />
        );
        expect(wrapper).toMatchSnapshot();
    });
});
