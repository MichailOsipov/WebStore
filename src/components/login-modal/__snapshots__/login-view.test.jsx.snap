// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginView rendering 1`] = `
<WithStyles(Grid)
  container={true}
  direction="column"
  spacing={16}
>
  <WithStyles(Grid)
    item={true}
  >
    <WithStyles(Typography)
      variant="h6"
    >
      Авторизация
    </WithStyles(Typography)>
  </WithStyles(Grid)>
  <WithStyles(Grid)
    item={true}
  >
    <ReduxForm />
  </WithStyles(Grid)>
  <WithStyles(Grid)
    item={true}
  >
    <WithStyles(Button)
      onClick={[MockFunction]}
      variant="contained"
    >
      Логин
    </WithStyles(Button)>
    <WithStyles(Button)
      onClick={[MockFunction]}
    >
      Назад
    </WithStyles(Button)>
  </WithStyles(Grid)>
</WithStyles(Grid)>
`;
